src/main/resources/templates/java-lang/javaClassGraphqlJacksonTypeIdResolver.ftl

```diff
@@ -0,0 +1,55 @@
+<#if package?has_content>
+package ${package};
+
+</#if>
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.DatabindContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.jsontype.impl.TypeIdResolverBase;
+
+<#if generatedAnnotation && generatedInfo.getGeneratedType()?has_content>
+@${generatedInfo.getGeneratedType()}(
+    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
+    date = "${generatedInfo.getDateTime()}"
+)
+</#if>
+public class GraphqlJacksonTypeIdResolver extends TypeIdResolverBase {
+
+    private JavaType superType;
+
+    @Override
+    public void init(JavaType baseType) {
+        superType = baseType;
+    }
+
+    @Override
+    public JavaType typeFromId(DatabindContext context, String typename) {
+        try {
+            var clazz = Class.forName(
+                <#if package?has_content>"${package}." +
+                </#if><#if modelNamePrefix?has_content>"${modelNamePrefix}" +
+                </#if>typename<#if modelNamePrefix?has_content> +
+                "${modelNameSuffix}"</#if>
+            );
```

Comment:
@gdhordain, sorry, I didn't get why do we need to consider `modelNameSuffix`/`modelNamePrefix` here?

Reply:
If the model name prefix and/or suffix is used, then the generated class name will use them whereas the graphql typename value will remains untouched.

In my test, typename is `UnionMemberA`, generated class name is `MyUnionMemberASuffix`.